uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/NavigationMenu/ConfigurationMenuModifier.php

```diff
@@ -0,0 +1,186 @@
+      'Insurance Plan Types' => 'hrjc_insurance_plantype',
+    ];
+
+    $this->insert($menu, $parent, $permission);
+
+    foreach ($optionGroupLinks as $itemName => $optionGroup) {
+      $link = 'civicrm/admin/options/' . $optionGroup . '?reset=1';
+      $path = $parent . '/' . $itemName;
+      $this->insert($menu, $path, $permission, $link);
+    }
+  }
+
+  /**
+   * Adds a submenu containing links to edit job role option groups
+   *
+   * @param array $menu
+   */
+  private function addJobRoleItems(&$menu) {
+    $parent = 'Administer/Job Roles';
+    $permission = 'Access CiviCRM';
+
+    $optionGroupLinks = [
+      'Locations' => 'hrjc_location',
+      'Regions' => 'hrjc_region',
+      'Departments' => 'hrjc_department',
+      'Levels' => 'hrjc_level_type',
+      'Cost Centres' => 'cost_centres',
+    ];
+
+    $this->insert($menu, $parent, $permission);
+
+    foreach ($optionGroupLinks as $itemName => $optionGroup) {
+      $link = 'civicrm/admin/options/' . $optionGroup . '?reset=1';
+      $path = $parent . '/' . $itemName;
+      $this->insert($menu, $path, $permission, $link);
+    }
+  }
+
+  /**
+   * Adds a submenu containing links to edit contact related option groups,
+   * and relationship types
+   *
+   * @param array $menu
+   */
+  private function addOtherStaffDetailsMenu(&$menu) {
+    $parent = 'Administer/Other Staff Details';
+    $permission = 'Access CiviCRM';
+
+    $this->insert($menu, $parent, $permission);
+
+    $optionGroupLinks = [
+      'Prefixes' => 'individual_prefix',
+      'Emergency Contact Relationships' => 'relationship_with_employee_20150304120408',
+      'Career History' => 'occupation_type_20130617111138',
+      'Disability Types' => 'type_20130502151940',
+      'Qualifications – Skill Categories' => 'category_of_skill_20130510015438',
+      'Qualifications – Skill Levels' => 'level_of_skill_20130510015934',
+    ];
+
+    foreach ($optionGroupLinks as $itemName => $optionGroup) {
+      $link = 'civicrm/admin/options/' . $optionGroup . '?reset=1';
+      $path = $parent . '/' . $itemName;
+      $this->insert($menu, $path, $permission, $link);
+    }
+
+    // add shortcut to relationship types edit
+    $relationshipTypesLink = 'civicrm/admin/reltype?reset=1';
+    $path = $parent . '/' . 'Manager Types';
+    $this->insert($menu, $path, $permission, $relationshipTypesLink);
+  }
+
+  /**
+   * Adds links to edit option groups related to leave and absence
+   *
+   * @param array $menu
+   */
+  private function addLeaveMenuItems(&$menu) {
+    $parent = 'Administer/leave_and_absences';
+    $permission = 'administer leave and absences';
+
+    $optionGroupLinks = [
+      'Sickness Reasons' => 'hrleaveandabsences_sickness_reason',
+      'TOIL to be Accrued' => 'hrleaveandabsences_toil_amounts',
+      'Work Pattern Change Reasons' => 'hrleaveandabsences_work_pattern_change_reason',
+      'Work Pattern Day Equivalents' => 'hrleaveandabsences_leave_days_amounts'
+    ];
```

Comment:
It's not clear to me why are these items being added dynamically. Is this a requirement from the specification? If it is not, then why don't we add them the same way we add the other items of the leave and absences menu?

Besides, I think that it's a bit confusing that some items will be added by hrcore and others by L&A.

Note that I'm talking about L&A here, but this actually applies to the other modifications done by this class

Reply:
When I started on this ticket I asked in the backend room about the choice between using dynamic inserts and database inserts. 

![image](https://user-images.githubusercontent.com/6374064/37156672-a9fa625e-22de-11e8-9969-566012c9e33c.png)

I got the impression that we were favoring dynamic inserts, although now I'm not so sure. Since these are all new submenus, and not alterations of core items, I think we can probably add them to the database, as long as you agree it makes sense too.